### PR TITLE
COMP: Update C++ source files removing code specific to Qt 4

### PR DESCRIPTION
**Pull request by jamesobutler**
This is a followup to https://github.com/commontk/CTK/commit/b86197b7ca9fd79c93702a114d87321de7229e7a

I used search terms such as `QT_VERSION <` and `QT_VERSION >` to find version checks beginning with `0x` to find more cases of Qt4 related logic to drop.

### CODE REVIEW

File: Libs/Widgets/Resources/ctkWidgets.qrc [hunk 1]
```diff
@@ -5,8 +5,6 @@
         <file>Icons/unlock.png</file>
         <file>Icons/minus.png</file>
         <file>Icons/plus.png</file>
-        <file>Icons/expand-down.png</file>
-        <file>Icons/expand-up.png</file>
```
Comment (jcfr) [inline Libs/Widgets/Resources/ctkWidgets.qrc:9]: Consider adding a note in the commit message explaining why these can be removed.

File: Libs/Widgets/ctkDirectoryButton.h [hunk 2]
```diff
@@ -31,10 +31,6 @@
 #include "ctkWidgetsExport.h"
 class ctkDirectoryButtonPrivate;
 
-// QFileDialog::Options can be used since Qt 4.7.0 (QT_VERSION >= 0x040700)
-// it is disabled to support older Qt versions
-//#define USE_QFILEDIALOG_OPTIONS 1
```
Comment (jcfr) [inline Libs/Widgets/ctkDirectoryButton.h:36]: I suggest to perform this update in a follow-up pull request as it does not strictly remove code.

Comment (jamesobutler) [reply Libs/Widgets/ctkDirectoryButton.h:36]: I've pushed a separate commit as `ENH: Remove custom CTK Dialog options`

File: Libs/Widgets/Resources/ctkWidgets.qrc (hunk 1 above)
Comment (jcfr) [reply Libs/Widgets/Resources/ctkWidgets.qrc:9]: Considering that local build of Slicer against this branch was :heavy_check_mark:, I will update the commit message like this and move forward with the integration:

```diff
- expand-down.png and expand-up.png are being removed as they were only used in ctkCollapsibleGroupBoxes built with Qt < 4.6
+ expand-down.png and expand-up.png are being removed as they were only used in ctkCollapsibleGroupBox built with Qt < 4.6
```

**Review by jcfr — APPROVED** [review]

**Pull request merged**